src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -525,6 +525,135 @@ public override BoundNode VisitTypeOrInstanceInitializers(BoundTypeOrInstanceIni
             return new BoundStatementList(node.Syntax, rewrittenStatements, node.HasErrors);
         }
 
+        public override BoundNode VisitArrayAccess(BoundArrayAccess node)
+        {
+            // If the array access index is of type System.Index or System.Range
+            // we need to translate it into a call into an actual indexer, instead
+            // of a simple array element access.
+
+            if (this.EmitModule is null || 
+                node.Indices.Length != 1 ||
+                node.Indices[0].Type.SpecialType != SpecialType.None)
```

Comment:
Shouldn't we be checking for the type being Index or Range instead of not a special type?